src/map.cpp

```diff
@@ -340,7 +341,7 @@ void Map::timerUpdate(float dtime, float unload_timeout, u32 max_loaded_blocks,
 	beginSave();
 
 	// If there is no practical limit, we spare creation of mapblock_queue
-	if (max_loaded_blocks == U32_MAX) {
+	if (max_loaded_blocks < 0) {
```

Comment:
This setting explicitly accepts the negative value `-1` which was then converted to an unsigned 32-bit integer. This change is the most straight-forward to handle all negative values instead of silently letting `-2` and alikes "underflow" during the cast. Even though that would result in a very high limit, the code should still skip unneeded calculations in that case and assume negative values are used to disable the limit.

Reply:
I see